exporter/datadog/datadog.go

```diff
@@ -0,0 +1,313 @@
+	var c config
+	for _, opt := range opts {
+		opt.apply(&c)
+	}
+
+	if c.statsAddr == "" {
+		c.statsAddr = defaultStatsAddr
+	}
+	if c.logger == nil {
+		c.logger = log.New(os.Stderr, "", log.LstdFlags)
+	}
+	if c.client == nil {
+		client, err := statsd.New(c.statsAddr)
+		if err != nil {
+			return config{}, err
+		}
+		c.client = client
+	}
+
+	return c, nil
+}
+
+// NewExporter returns a new Datadog stats Exporter
+func NewExporter(opts ...Option) (*Exporter, error) {
+	c, err := buildConfig(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		lastValues = newLastValues()
+		exporter   = Exporter{
+			lastValues: lastValues,
+			client:     c.client,
+			logger:     c.logger,
+		}
+	)
+
+	return &exporter, nil
+}
+
+// ExportView implements view.Exporter
+func (e *Exporter) ExportView(data *view.Data) {
+	go e.publish(data)
+}
+
+// publishCount publishes CountAggregation metrics via dogstatsd
+func (e *Exporter) publishCount(name string, data, previous view.AggregationData, tags []string, rate float64) {
+	var (
+		ptr   = data.(*view.CountData)
+		value = *ptr
+	)
+
+	if lastValue, ok := previous.(*view.CountData); ok {
+		value = value - *lastValue
+	}
+
+	if err := e.client.Count(name, int64(value), tags, rate); err != nil {
+		e.logger.Printf("unable to publish Count metric, %v\n", err)
+	}
+}
+
+// publishSum publishes SumAggregation metrics via dogstatsd
+func (e *Exporter) publishSum(name string, data, previous view.AggregationData, tags []string, rate float64) {
+	var (
+		ptr   = data.(*view.SumData)
+		value = *ptr
+	)
+
+	if lastValue, ok := previous.(*view.SumData); ok {
+		value = value - *lastValue
+	}
+
+	if err := e.client.Count(name, int64(value), tags, rate); err != nil {
+		e.logger.Printf("unable to publish Count metric, %v\n", err)
+	}
+}
+
+// publishMean publishes MeanAggregation metrics via dogstatsd
+func (e *Exporter) publishMean(name string, data view.AggregationData, tags []string, rate float64) {
+	var (
+		value = data.(*view.MeanData)
+	)
+
+	if err := e.client.Gauge(name, value.Mean, tags, rate); err != nil {
+		e.logger.Printf("unable to publish Gauge metric, %v\n", err)
+	}
+}
+
+// publishDistribution publishes DistributionAggregation metrics via dogstatsd
+func (e *Exporter) publishDistribution(name string, buckets view.DistributionAggregation, data, previous view.AggregationData, tags []string, rate float64) {
+	var (
+		value             = data.(*view.DistributionData)
+		length            = len(value.CountPerBucket)
+		lastValue, lastOk = previous.(*view.DistributionData)
+		lastLength        int
+	)
+
+	if lastOk {
+		lastLength = len(lastValue.CountPerBucket)
+	}
+
+	for index, bucket := range buckets {
+		var count int64
+		if index < length {
+			count = value.CountPerBucket[index]
+		}
+		if index < lastLength {
+			count = count - lastValue.CountPerBucket[index]
+		}
+
+		for i := int64(0); i < count; i++ {
+			e.client.Histogram(name, bucket, tags, rate)
```

Comment:
Seems odd that we make the same call repeatedly

Reply:
Yep, I thought so too.  statsd and Views are conceptually very similar.  They take raw events and roll them up into something interesting.  What I was looking for was effectively a way to undo what View had done.  

In an ideal world, I could supply a statsd view and allow statsd to do this.  As it is, I'm try to reconstruct the originating measures from the view.